manifest/v1alpha/slo/validation.go

```diff
@@ -184,6 +260,13 @@ var specCompositeObjectiveValidationRule = validation.NewSingleRule(func(s Spec)
 	return nil
 })
 
+var compositeObjectiveRule = validation.New[CompositeObjective](
+	validation.ForPointer(func(c CompositeObjective) *float64 { return c.Weight }).
+		WithName("weight").
+		Required().
+		Rules(validation.GreaterThan(0.0)),
```

Comment:
Fields `project`, `slo`, `objective` and `whenDelayed` are required. Additionally, we could validate if `project`, `slo`, `objective` are a proper subdomains, using `validation.StringIsDNSSubdomain()`. `project` and `name` are originally validated with `validationV1Alpha.FieldRuleMetadataName` and `validationV1Alpha.FieldRuleMetadataProject` but I am not sure if we should reuse those.

Reply:
Added, with tests